crates/hir-ty/src/infer/closure.rs

```diff
@@ -263,6 +269,103 @@ impl CapturedItem {
         self.span_stacks.iter().map(|stack| *stack.last().expect("empty span stack")).collect()
     }
 
+    /// Converts the place to a name that can be inserted into source code.
+    pub fn place_to_name(&self, owner: DefWithBodyId, db: &dyn HirDatabase) -> String {
+        use std::fmt::Write;
+
+        let body = db.body(owner);
+        let mut result = body[self.place.local].name.unescaped().display(db.upcast()).to_string();
+        for proj in &self.place.projections {
+            match proj {
+                ProjectionElem::Deref => {}
+                ProjectionElem::Field(Either::Left(f)) => {
+                    match &*f.parent.variant_data(db.upcast()) {
+                        VariantData::Record(fields) => {
+                            result.push('_');
+                            result.push_str(fields[f.local_id].name.as_str())
+                        }
+                        VariantData::Tuple(fields) => {
+                            let index = fields.iter().position(|it| it.0 == f.local_id);
+                            if let Some(index) = index {
+                                write!(result, "_{index}").unwrap();
```

Comment:
We have a `format_to` macro in stdx for this (that will drop the unwrap and `Write` import)